### PR TITLE
Allow update from Lightbend SBR in ClusterReceptionistConfigCompatChecker

### PR DESCRIPTION
**Pull request by patriknw**
* and don't allow different strategies

### CODE REVIEW

File: akka-cluster/src/main/scala/akka/cluster/sbr/SplitBrainResolverProvider.scala [hunk 1]
```diff
@@ -61,3 +61,44 @@ final class SplitBrainResolverProvider(system: ActorSystem) extends DowningProvi
   }
 
 }
+
+final class SplitBrainResolverProvider2(system: ActorSystem) extends DowningProvider {
```
Comment (chbatey) [inline akka-cluster/src/main/scala/akka/cluster/sbr/SplitBrainResolverProvider.scala:65]: is this meant to be in?

File: akka-cluster/src/main/scala/akka/cluster/JoinConfigCompatCheckCluster.scala [hunk 2]
```diff
@@ -9,15 +9,44 @@ import scala.collection.{ immutable => im }
...
 final class JoinConfigCompatCheckCluster extends JoinConfigCompatChecker {
+  import JoinConfigCompatCheckCluster._
+
+  override def requiredKeys: im.Seq[String] = List(DowningProviderPath, SbrStrategyPath)
+
+  override def check(toCheck: Config, actualConfig: Config): ConfigValidation = {
+    val toCheckDowningProvider = toCheck.getString(DowningProviderPath)
+    val actualDowningProvider = actualConfig.getString(DowningProviderPath)
+    val downingProviderResult =
+      if (toCheckDowningProvider == actualDowningProvider || Set(toCheckDowningProvider, actualDowningProvider) == Set(
```
Comment (chbatey) [inline akka-cluster/src/main/scala/akka/cluster/JoinConfigCompatCheckCluster.scala:38]: 👍

File: akka-cluster/src/main/scala/akka/cluster/sbr/SplitBrainResolverProvider.scala (hunk 1 above)
Comment (patriknw) [reply akka-cluster/src/main/scala/akka/cluster/sbr/SplitBrainResolverProvider.scala:65]: wow, no, it was just some experiments

**Review by johanandren — APPROVED** [review]

**Review by octonato — APPROVED** [review]
LGMT

Amazing how I completely forgot how the `JoinConfigCompactCheck`. Good piece of software! ;-)

**Pull request merged**